benchmark/Cajita_SparsePartitionerPerformance.cpp

```diff
@@ -0,0 +1,457 @@
+/****************************************************************************
+ * Copyright (c) 2018-2021 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "Cabana_BenchmarkUtils.hpp"
+
+#include <Cajita_SparseDimPartitioner.hpp>
+#include <Cajita_SparseIndexSpace.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <set>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <mpi.h>
+
+//---------------------------------------------------------------------------//
+// Helper functions.
+struct ParticleWorkloadTag
+{
+};
+struct SparseMapTag
+{
+};
+
+std::set<std::array<int, 3>> generateRandomTiles( int tiles_per_dim,
+                                                  int tile_num )
+{
+    std::set<std::array<int, 3>> tiles_set;
+    while ( static_cast<int>( tiles_set.size() ) < tile_num )
+    {
+        std::array<int, 3> rand_tile;
+        for ( int d = 0; d < 3; ++d )
+            rand_tile[d] = std::rand() % tiles_per_dim;
+        tiles_set.insert( rand_tile );
+    }
+    return tiles_set;
+}
+
+// generate a random tile sequence
+int current = 0;
+int uniqueNumber() { return current++; }
+
+Kokkos::View<int* [3], Kokkos::HostSpace>
+generateRandomTileSequence( int tiles_per_dim ) {
```

Comment:
Does clang-format hit these files? We usually don't have `{` trailing a function definition on the same line.